src/api/users.ts

```diff
@@ -0,0 +1,31 @@
+import axios from '../libs/axios';
+import { AxiosPromise } from 'axios';
+import {
+    Post,
+    Series,
+    Question,
+    UserItem,
+    TagItem,
+    SocialAccount,
+    Request,
+    PaginatedResponse
+} from '../types/api';
+
+export interface Profile extends UserItem {
+    social_accounts: Array<SocialAccount>;
+}
+
+export const getProfile = (username: string, params?: object): Promise<Profile> =>
+    axios.get(`/users/${username}`, { params }).then(_ => _.data);
```

Comment:
@framgia-phuongth How many GET routes don't use this? 😄 
What do you think about adding an interceptor for all GET requests and fix API for those which don't use this pattern?

Reply:
Wait. Never mind. I confused with response.data.data